src/sort.rs

```diff
@@ -262,9 +321,51 @@ fn sort_by_ordering(
 ) {
     let mut idx = 0;
     for heading in ordering {
-        if let Some((_, to_sort_headings)) =
-            heading_order.iter().find(|((_, key), _)| key == heading)
-        {
+        let mut matches: Vec<(&(usize, String), &Vec<Heading>)> = heading_order
+            .iter()
+            .filter(|((_s, key), headings)| {
+                key == heading
+                    || headings.iter().any(|h| {
+                        if let Heading::Complete(segs) = h {
+                            segs.iter().any(|seg| seg == heading)
+                        } else {
+                            false
+                        }
+                    })
+            })
+            .collect();
+
+        matches.sort_by(|a, b| {
+            let a1_longest =
+                a.1.iter()
```

Comment:
Can we destruct `a` and `b` here into their tuples to make this a bit easier to read?

Reply:
This closure is only used in this one place, and it would be unintuitive to separate it into a separate function.

Reply:
Why not:

```rust
matches.sort_by(|((?, ?), ?), ((?, ?), ?)| {
```

And give the variables meaningful names?